10 - Footer Component - Khadija

KhadijaHaji opened the pull request:
## Description

This PR Creates the first iteration of the Footer using the first iteration Wireframes.


-  You can see address, contact details, services offered and a list of pages.
-  In mobile first that it will be a single column of 4
-  In desktop version one row of 4 columns


## Related to


issue #10 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation - Not currently necessary

KhadijaHaji committed:
Add address to Footer component

KhadijaHaji committed:
adds services to footer

KhadijaHaji committed:
adds pages to Footer component

KhadijaHaji committed:
adds background color and padding

KhadijaHaji committed:
styled the foot-container so it's in column

KhadijaHaji committed:
media query for mobile first and desktop

KhadijaHaji committed:
changed background color to grey and got rid of list styling

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,58 @@
+footer {
+    background-color:#B3B3B3;;
+    padding: 20px;
+}
+
+.footer-container {
+    display: flex;
+    flex-direction: column;
+}
```
```suggestion
footer {
    display: flex;
    flex-direction: column;
    gap: 10px; /* or whatever gap you want */
    padding: 20px 20px 40px 20px;
    background-color:#B3B3B3;
}
```
we have a footer and then a div inside of it which are fulfilling the same function and can be combined into one

but maybe there is another reason that im missing

there are some adjustments suggestions here that are relative to the comments below

KhadijaHaji replied:
I understand what you mean, I'll commit your suggestion thanks! @bazmurphy

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,58 @@
+    color: white;
+}
+
+.address,
+.contact-details,
+.services,
+.pages {
+    margin-bottom: 20px;
+}
+
+.address {
+    margin-bottom: 10px;
+}
```
is this to offset each item from one another in the mobile first column mode? 
in that case we can use gap (see above and below)

or is this to provide padding at the bottom of the entire footer container in desktop? 
in which case we can add that to the footer element (see above)

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,58 @@
+
+.no-list-style {
+    list-style-type: none;
+    padding: 0;
+    margin: 0;
```
everything has `padding: 0` and `margin: 0` globally applied to it by the `RootLayout.css` which might eventually become a global css stylesheet

KhadijaHaji replied:
great, thanks, I've removed these now.

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,58 @@
+/* Mobile-first design */
+@media (min-width: 768px) {
+
+    /* Desktop version */
+    .footer-container {
+        flex-direction: row;
+        flex-wrap: wrap;
+    }
```
```suggestion
    /* Desktop version */
    footer {
        flex-direction: row;
        justify-content: space-evenly;
        gap: 20px; /* or whatever gap you want */
    }
```
we can use this on the footer if we apply the above logic.
the `gap` will change according to mobile/desktop
the `justify-content` prevents the need for the `flex-basis` percentages

and about the `flex-wrap`, is the thinking when the screen gets too small for 4 columns and you want it to snap over underneath? this is a good thing for us to think about re: breakpoints.

just my own thoughts (for future):
maybe grid would be better then eg.
1 col x 4 row in mobile
2 col x 2 row in tablet
4 col x 1 row in desktop

KhadijaHaji replied:
I've committed the changes you've suggested.
The first point you made about flex-wrap is why I used it.

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,58 @@
+        flex-wrap: wrap;
+    }
+
+    .address,
+    .contact-details,
+    .services,
+    .pages {
+        flex-basis: 25%;
+    }
+}
```
percentages get funky with gaps

because it's trying to fill 100% of the container and the `gap`s in between cause a problem with that which means you end up with overspill

see above where i added `justify-content: space-evenly` on the desktop version
and you could maybe specify another `justify-content` on the mobile to be explicit if you wanted

bazmurphy commented on client/src/components/Footer/Footer.jsx:
```diff
@@ -1,11 +1,86 @@
+import React from "react";
```
we don't need this import statement in modern versions of react anymore

bazmurphy commented on client/src/components/Footer/Footer.jsx:
```diff
@@ -1,11 +1,86 @@
+      <div className="footer-container">
+        <div className="address">
+          <h3 className="footer-headings">ADDRESS</h3>
+          <p>Lorem Ipsum is simply dummy text of the printing</p>
```
There is a way to edit your VSCode settings so you can use "Emmet" in jsx files, so your intellisense will work when you type `lorem10` and then you press enter and it gives you a 10 word sentence of lorem, this might be useful for you on this project because there's going to be a lot of placeholder text 🤣

KhadijaHaji replied:
IT WORKS🤣

bazmurphy commented on client/src/components/Footer/Footer.jsx:
```diff
@@ -1,11 +1,86 @@
+          <h3 className="footer-headings">SERVICES</h3>
+          <ul className="no-list-style">
+            <li>
+              <a href="#">Service 1</a>
```
Will these `<a>` elements eventually link to something like:

`/services#1`

as in the `id` of the heading of each service from the `ServicesPage`?

maybe worth pre-populating them to avoid the a11y eslint accessibility error

I will look up how to achieve the above id thing with `Link` from React Router, it might be simple...

KhadijaHaji replied:
I've linked them to the ServicesPage for now then I will populate it with the id of each heading of the ServicePage.

bazmurphy commented on client/src/components/Footer/Footer.jsx:
```diff
@@ -1,11 +1,86 @@
+          <h3 className="footer-headings">PAGES</h3>
+          <ul className="no-list-style">
+            <li>
+              <NavLink to="/">Home</NavLink>
```
Just to explain the difference between `Link` and `NavLink` from React Router.
The `NavLink` has an automatically applied class called "active" when the route of the current page matches it

If you would like that benefit we can keep these as `NavLinks` otherwise they can be `Link`
But this is not a big thing

KhadijaHaji replied:
I've changed the NavLink to Link.

bazmurphy reviewed (changes requested):
Hi @KhadijaHaji  😄 🥳 

Here is my Review

These are my suggestions, and its a discussion, I am happy to communicate on these points and clarify anything, and please stand your ground if you are doing things for specific reasons, just try to explain it etc.

Please have a play with the suggestions and see how they change things.
And I recommend (whilst working) using this line **temporarily**, because it really helps you to see whats happening with all the box models and why (see attached image)
```
* {
  border: 1px dotted black;
}
```
![image](https://github.com/ShayanMahnam/lentegeur-hospital-facility-board/assets/61154071/418b6af4-bbf1-4257-9d43-119c7f9e1666)

Please mention me like @bazmurphy when replying so I get notifications

(Also you can edit the PR description to check each of the checkboxes, by putting an x inside each [ ], you can double check on the Preview tab when you are trying to do it)

Thanks!

KhadijaHaji committed:
Update client/src/components/Footer/Footer.css

Co-authored-by: Baz <61154071+bazmurphy@users.noreply.github.com>

KhadijaHaji committed:
Update client/src/components/Footer/Footer.css

Co-authored-by: Baz <61154071+bazmurphy@users.noreply.github.com>

KhadijaHaji committed:
edited css

removed padding/margin from list-style.
added    justify-content: space-evenly to headings

ShayanMahnam commented on client/src/components/Footer/Footer.jsx:
```diff
@@ -1,11 +1,85 @@
-	);
+  return (
+    <footer>
+      <div className="footer-container">
```
if you delete this div (footer-container) everything will work perfectly

KhadijaHaji replied:
Thanks @ShayanMahnam , I've removed it.

KhadijaHaji committed:
worked on the review comments

KhadijaHaji committed:
changed editor format back to false on save

KhadijaHaji committed:
added a max width of 1200px

KhadijaHaji committed:
max width added to mobile version - 1200px

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,56 @@
+    flex-direction: column;
+    gap: 20px;
+    padding: 30px 20px 45px 20px;
+    font-size: 12px;
```
instead of explicitly stating this font-size
it is probably better to leave this font sizing to the global style sheet 
where in mobile it will be 14px 
and in desktop it will be 16px 
and so will be consistent with the rest of the text on the page

unless of course you specifically want it smaller

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,56 @@
+}
+
+.footer-subheading {
+    font-size: 15px;
```
same as above re: font sizes

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,56 @@
+
+    .footer-subcontainer {
+        flex-basis: 25%;
+        gap: 10px;
```
this is the gap between the heading and the list in Desktop and its currently 10px
but it's smaller than the one for Mobile (currently 15px)

so either we can make them the same both 10px or both 15px (which you like)

or have the mobile less and the desktop more (which makes visual sense)

up to you

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,56 @@
+.footer-subcontainer {
+    display: flex;
+    flex-direction: column;
+    gap: 15px;
```
see other comment re: gap in the media query

this is Mobile and the gap is currently 15px (5px larger than the Desktop gap)

bazmurphy commented on client/src/components/Footer/Footer.css:
```diff
@@ -0,0 +1,56 @@
+footer {
+    background-color: hsl(0, 0%, 70%);
```
we can use the global variables now
```suggestion
    background-color: var(--gray70);
```

bazmurphy replied:
(don't commit this suggestion using the button above, just change it in your own code)

KhadijaHaji replied:
Great, all done.

bazmurphy reviewed (approved):
Good job 😎 It has improved nicely 😄
(Just a few minor comments to address)


1. Please make sure you enable that Prettier setting (locally - do not commit it) to format your code (the tabs should be 2 spaces etc)

2. Please make sure you update this branch with the latest `development` branch **before** you merge the PR

- switch to the development branch
  - `git switch development`
  - get the latest version of development
  - `git fetch origin`
  - `git pull origin development`

- then switch back to your branch
  - `git switch 10-footer-component`
  - and merge the latest version of development into your branch
  - `git merge development`
  - and it will add that as a (merge) commit to your branch
  
- and finally add, commit and push (like normal)...

And you can then merge this PR 👍

KhadijaHaji committed:
Merge remote-tracking branch 'origin/development' into 10-footer-component

KhadijaHaji committed:
added padding to mobile version

KhadijaHaji committed:
replaced hsl background color to the global css variable
